### PR TITLE
[CAO-22200] - new capability of testing with secure identity

### PR DESCRIPTION
**Pull request by powerbot15**
Added secure identity and improved SPA page to have ability to check secure and async identity. Changes are mostly around adding one more checkbox for regular and spa pages for all envs

image

### CODE REVIEW

File: v2/js/spa.js [hunk 1]
```diff
@@ -160,18 +204,18 @@
...
             lpTag.identities.push(getAuthData);
             lpTag.sdes.push({type: 'ctmrinfo', info: {customerId: 'lpTest' + userName}});
-
-            window.LPJsMethodName = function (callback) {
-                callback(userName);
-            };
-
-            window.LPGetAuthenticationToken = function (callback) {
-                callback(userName);
+            window.LPJsMethodName = window.LPGetAuthenticationToken = function (callback) {
```
Comment (jmcleanlp) [inline v2/js/spa.js:213]: I'm not a big fan of setting values this way but it's not a huge deal

**Review by jmcleanlp — APPROVED** [review]
If the timeout thing is not going to be an issue I think this is good

File: v2/js/spa.js [hunk 2]
```diff
@@ -183,35 +227,49 @@
...
+                acr: 'loa1'
+            }
+            if (app.data.isSecureIdentity) {
+                authData.tkn = userName;
+                authData.redirect_uri = window.location.href;
+            } else {
+                authData.sub = userName;
+            }
+            if (app.data.isAsyncIdentity) {
+                setTimeout(function () {
```
[the remark this answers is not in the document]
Comment (powerbot15) [reply v2/js/spa.js:247]: @jmcleanlp It shouldn't cause any issues as it is just a simulation of a network request to brand's IDP if they generating auth data via API for example. We even had an incident in the past on auth flow as an async process was not considered in some deployed code

File: v2/js/spa.js (hunk 1 above)
Comment (powerbot15) [reply v2/js/spa.js]: Segregated and moved to a separate method

**Pull request merged**